### DISCUSSION TITLE
Language plugin recommendations

### DISCUSSION THREAD

**Discussion by robertmoura** [malobre/dprint-plugin-vue#8] [Q&A]
Love the idea of being able to configure all the languages in a Vue SFC file separately! Such a great idea. Had a bit of trouble finding good plugins to format Vue's style of HTML though. CSS or SCSS wasn't easy either. Did you have any recommendations to help format these languages well?

Here's a snippet of the configuration I'm using now:

```json
{
    // ...omitted...
    "prettier": {
        "associations": [
            "*.{css,scss}"
        ]
    },
    "includes": [
        "*.{ts,js,cjs,mjs,json,vue,css,scss}"
    ],
    "excludes": [
        "**/node_modules",
        "**/*-lock.json"
    ],
    "plugins": [
        "https://plugins.dprint.dev/typescript-0.86.2.wasm",
        "https://plugins.dprint.dev/json-0.17.4.wasm",
        "https://plugins.dprint.dev/malobre/vue-v0.5.0.wasm",
        "https://plugins.dprint.dev/prettier-0.27.0.json@3557a62b4507c55a47d8cde0683195b14d13c41dda66d0f0b0e111aed107e2fe"
    ]
}
```

Things I've tried for HTML:
- https://github.com/dprint/dprint-plugin-prettier
  For the HTML, I tried Prettier but because it has different formatting options for Vue files verses HTML files it treats it a bit differently. To get around this I also tried passing the html as a Vue file by setting `parser` to `"vue"` but it expects a template tag to be wrapped around the HTML to be formatted properly.

What I've tried for CSS/SCSS:
- https://github.com/disrupted/dprint-plugin-css
  This plugin didn't seem to work well with CSS (nesting wasn't supported). I don't think it has an option to allow SCSS either.
- https://github.com/dprint/dprint-plugin-prettier
  This works well but seems overkill and is not super configurable. Although better than nothing!

**Comment by malobre** [comment]
I am not actively using dprint anymore, I have no plugin recommendations.
